informer/integration/test/deletionerror/deletion_error_test.go

```diff
@@ -0,0 +1,129 @@
+	}
+
+	// When there are the config maps in the Kubernetes API we start the watch.
+	deleteChan, updateChan, errChan := operatorkitInformer.Watch(ctx)
+
+	// We define a general control goroutine to stop test execution on errors or
+	// after timeouts. The timeout is 25 seconds because of two resync periods,
+	// one rate limit wait period, plus some time buffer for all the test magic
+	// happening here: (2 * 10) + 2 + buffer.
+	go func() {
+		for {
+			select {
+			case <-time.After(25 * time.Second):
+				t.Fatalf("expected proper test execution got timeout")
+			case err := <-errChan:
+				if err != nil {
+					t.Fatalf("expected %#v got %#v", nil, err)
+				}
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+
+	// Due to the two created config maps we get two events in the informer which
+	// we simply drain here because they stand in our way of testing the actual
+	// matter of delete events. When anything else is weird with the informer and
+	// the channels hang on this stage then the test will time out due to the
+	// safety goroutine above.
+	<-updateChan
+	<-updateChan
+
+	// We disable the event dispatching in the filter watcher immediately before
+	// any config maps are created in the Kubernetes API.
+	filterWatcher.SetDispatchEvents(false)
+
+	// Now we delete a the first config map. This event is expected to be received
+	// immediately in normal circumstances. For this test though, we disabled
+	// event dispatching in the filter watcher. Thus we check if there was any
+	// event at all. Note that we sleep a bit after deleting the config map to let
+	// the delete event go through the whole stack. This may takes a while so we
+	// simply set it to 5 seconds.
+	{
+		err := deleteConfigMap(idOne)
+		if err != nil {
+			t.Fatal("expected", nil, "got", err)
+		}
+
+		time.Sleep(5 * time.Second)
+
+		select {
+		case <-deleteChan:
+			t.Fatalf("did not expect any event but got delete event")
+		case <-updateChan:
+			t.Fatalf("did not expect any event but got update event")
+		default:
+			// fall through
+		}
+	}
+
+	// Now the first config map got deleted and we ignored the delete event. The
+	// informer did not have any chance to perceive the deletion on its own. We
+	// enable the event dispatching again to check what config maps the informer
+	// still knows about.
+	filterWatcher.SetDispatchEvents(true)
+
+	// After enabling the event dispatching in the filter watcher again, we should
+	// start to receive events again from the informer. Since the first config map
+	// got deleted in the Kubernetes API, we should only receive an update event
+	// anymore, due to the second config map. Note that we loop over the event
+	// channels because of a potential bug that could make the test succeed
+	// occasionally. In case the filter watcher is buggy and dispatches events
+	// even though we do not want or expect that, there would be two config maps
+	// dispatched through the informer. The order of these two objects would never
```

Comment:
I would like to keep it. When I come back in 6 months I will be glad I wrote it down. Let's start out with that. When we think this is useless in a year because it did not help we can still remove it. But so far I like to track some more detailed background information.